packages/realm-server/tests/atomic-endpoints-test.ts

```diff
@@ -0,0 +1,788 @@
+          let place1Response = await request
+            .get('/place-modules/place1.gts')
+            .set('Accept', 'application/vnd.card+source');
+          assert.strictEqual(place1Response.status, 200, 'HTTP 200 status');
+          assert.strictEqual(
+            place1Response.get('X-boxel-realm-url'),
+            testRealmHref,
+            'realm url header is correct',
+          );
+          assert.strictEqual(
+            place1Response.get('X-boxel-realm-public-readable'),
+            'true',
+            'realm is public readable',
+          );
+          assert.strictEqual(
+            place1Response.text.trim(),
+            place1Source,
+            'the card source is correct',
+          );
+          let place2Response = await request
+            .get('/place-modules/place2.gts')
+            .set('Accept', 'application/vnd.card+source');
+          assert.strictEqual(place2Response.status, 200, 'HTTP 200 status');
+          assert.strictEqual(
+            place2Response.text.trim(),
+            place2Source,
+            'the card source is correct',
+          );
+          assert.strictEqual(
+            place2Response.get('X-boxel-realm-url'),
+            testRealmHref,
+            'realm url header is correct',
+          );
+          assert.strictEqual(
+            place2Response.get('X-boxel-realm-public-readable'),
+            'true',
+            'realm is public readable',
+          );
+
+          let countryResponse = await request
+            .get('/country.gts')
+            .set('Accept', 'application/vnd.card+source');
+          assert.strictEqual(countryResponse.status, 200, 'HTTP 200 status');
+          assert.strictEqual(
+            countryResponse.text.trim(),
+            countrySource,
+            'the card source is correct',
+          );
+          assert.strictEqual(
+            countryResponse.get('X-boxel-realm-url'),
+            testRealmHref,
+            'realm url header is correct',
+          );
+          assert.strictEqual(
+            countryResponse.get('X-boxel-realm-public-readable'),
+            'true',
+            'realm is public readable',
+          );
+        });
+        test('can write a single instance', async function (assert) {
+          let doc = {
+            'atomic:operations': [
+              {
+                op: 'add',
+                href: '/new-person-1.json',
+                data: {
+                  type: 'card',
+                  attributes: {
+                    firstName: 'Mango',
+                  },
+                  meta: {
+                    adoptsFrom: {
+                      module: '/person',
+                      name: 'Person',
+                    },
+                  },
+                },
+              },
+            ],
+          };
+          let response = await request
+            .post('/_atomic')
+            .set('Accept', 'application/vnd.card+json')
```

Comment:
Does this Accept header make sense for posting to this endpoint? You aren't getting card json back are you?

Reply:
Indeed